src/Linters/NoNewlineAtStartOfControlFlowBlockLinter.hack

```diff
@@ -0,0 +1,78 @@
+/*
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\HHAST;
+
+use namespace HH\Lib\Str;
+
+class NoNewlineAtStartOfControlFlowBlockLinter extends AutoFixingASTLinter {
+  const type TNode = IControlFlowStatement;
+  const type TContext = Script;
+
+  <<__Override>>
+  public function getLintErrorForNode(
+    Script $_context,
+    IControlFlowStatement $node,
+  ): ?ASTLintError {
+    $body = $this->getBody($node);
+    if (!$body is CompoundStatement) {
+      return null;
+    }
+
+    $left_brace = $body->getLeftBrace();
+    $next_token = $this->getAST()->getNextToken($left_brace);
+
+    if ($next_token is null) {
+      return null;
+    }
+
+    $whitespace = $next_token->getLeading()->getCode();
```

Comment:
This seems to rely on the assumption that the first EOL will be part of the left brace's trailing trivia and any other EOLs will be part of the next token's leading trivia.

I guess that's probably true right now but I'm not sure if we can rely on that never changing...it might be more robust to do something like this here:

```Hack
$whitespace =
  $left_brace->getTrailing()->getCode().
  $next_token->getLeading()->getCode();
```

(then check for double EOL at the beginning)

BTW do we have/should we have a linter for empty lines at the *end* of a control block?

Reply:
That would not work in case of
```HACK
if(true){//comment

  f();
}
```

If this hidden knowledge is subject to change, we can accept this false negative, right?

Reply:
I don't really have an opinion on whether that should count as a false positive/negative, but either way, it shouldn't be too hard to implement whichever exact check we decide on using either the concatenated getCode() strings, or the concatenated list of trailing+leading trivia.

Another edge case I can think of would be something like

```Hack
if (true) { /* multi-line
  comment */

  f();
}
```

and there might be others...

Reply:
I added these test cases. Current AST treats them correctly. If the AST changes and hands the eol to the next node, this continues to work. If all trivia is handed to the next node, only pure eol cases will continue to work. Those are the most common anyway. The new autofix doesn't force all code on the same line as the curly if the AST were to shift. I think this state is a good tradeoff between code complexity now and correct behavior later.